gr-blocks/lib/moving_average_XX_impl.cc.t

```diff
@@ -95,18 +101,37 @@ namespace gr {
       const @I_TYPE@ *in = (const @I_TYPE@ *)input_items[0];
       @O_TYPE@ *out = (@O_TYPE@ *)output_items[0];
 
-      @I_TYPE@ sum = 0;
-      int num_iter = (noutput_items>d_max_iter) ? d_max_iter : noutput_items;
-      for(int i = 0; i < d_length-1; i++) {
-        sum += in[i];
-      }
-
-      for(int i = 0; i < num_iter; i++) {
-        sum += in[i+d_length-1];
-        out[i] = sum * d_scale;
-        sum -= in[i];
+      unsigned int num_iter = (unsigned int)((noutput_items>d_max_iter) ? d_max_iter : noutput_items);
+      if(d_vlen == 1) {
+        @I_TYPE@ sum = 0;
+        for(unsigned int i = 0; i < d_length-1; i++) {
+          sum += in[i];
+        }
+
+        for(unsigned int i = 0; i < num_iter; i++) {
+          sum += in[i+d_length-1];
+          out[i] = sum * d_scale;
+          sum -= in[i];
+        }
+
+      } else { // d_vlen > 1
+        //TODO: VOLKize this zeroing?
+        std::fill(d_sum.begin(), d_sum.end(), (@I_TYPE@)0);
```

Comment:
Having spent some time with the C++ standard library, I don't trust these algorithms to be as high-performance as low-level system calls. Although I hate mixing C and C++ calls, a memset here will will have a better chance of being optimal performance.